test/environmentCreate/SpecificationCreate.test.tsx

```diff
@@ -48,20 +71,40 @@ describe("<SpecificationCreate />", () => {
     const switchButton = component.getByLabelText("YAML", { exact: false });
     fireEvent.click(switchButton);
 
+    expect(
+      screen.getByText((content, element) => {
+        return (
+          element?.textContent === "channels:  -dependencies:  -variables: {}"
+        );
+      })
+    );
+
```

Comment:
```suggestion
    const textbox = await component.findByText("channels: -dependencies: -variables: {}");
    expect(textbox).toBeInDocument();
```

I assume this will give a better error message.

JavaScript developers are not used to seeing an unchained/naked expect() call (e.g. `toBeInDocument`). 

A JS dev would have to know that `getByText()` throws an error, and that's essentially what will cause this test to fail, rather than the chained call to expect() failing (and throwing).

For reference, [testing-library about queries](https://testing-library.com/docs/queries/about#types-of-queries)

Reply:
I'll try to rewrite these using the suggestion above, but here's some context for why the current code is written the way it is.

First, this screenshot, which I took while developing this, shows how the HTML code looks like (this is the case when `variables` is empty, set to `{}`):

<img width="332" alt="Screen Shot 2024-02-11 at 16 25 57" src="https://github.com/conda-incubator/conda-store-ui/assets/10469885/4080aed5-fe5b-4415-bc32-4778f66a05c5">

So my initial idea was to search for a substring (like `variables`, which is inside the `span` tag) and then look at the parent node, which is inside the `div`, and check the string there. However, I then discovered this discussion:

https://stackoverflow.com/questions/62770973/parent-node-in-react-testing-library

with this top answer:

```js
screen.getByText("span text").closest("div")
```

What I don't like about this is that this doesn't establish a relationship to the element I'm matching on. It's just the closest div, which could be elsewhere.

There's also a discussion there on what's considered good practice in the react testing library and people say there are even default eslint rules to enforce certain things. And it might be not a good idea to access parent elements directly.

So I wanted to find something else that would allow me to capture the entire textual output of the YAML.

The only thing I found was `textContent`. But how can I get that? I looked for a suggestion again and came across the `getByText((content, element) => ...` overload, probably in this discussion:

https://stackoverflow.com/questions/61675320/how-to-get-the-child-div-of-parent-div-with-some-test-id-using-react-testing-lib

So I just ended up using this with `textContent`.

One thing I definitely need to improve is to match on `component` instead of the entire document.

----

@gabalafou While I was typing this reply, I actually found another suggestion here:

https://stackoverflow.com/questions/55509875/how-to-query-by-text-string-which-contains-html-tags-using-react-testing-library

So I wonder if this would be better (it still allows to match on `textContent` and takes advantage of the `.to` methods of the testing library):

```js
const el = getByText('Name:', { exact: false })
expect(el.textContent).toEqual('Name: Bob (special guest)');
```